gfx/drivers_shader/slang_process.cpp

```diff
@@ -0,0 +1,406 @@
+            {
+               uniform.offset = src.push_constant_offset;
+               uniforms[SLANG_CBUFFER_PC].push_back(uniform);
+            }
+            else if (src.uniform)
+            {
+               uniform.offset = src.ubo_offset;
+               uniforms[SLANG_CBUFFER_UBO].push_back(uniform);
+            }
+         }
+      }
+      texture_map++;
+   }
+
+   out->texture_count = textures.size();
+
+   textures.push_back({ NULL });
+   out->textures = (texture_sem_t*)malloc(textures.size() * sizeof(*textures.data()));
+   memcpy(out->textures, textures.data(), textures.size() * sizeof(*textures.data()));
+
+   for (int i = 0; i < SLANG_CBUFFER_MAX; i++)
+   {
+      if (uniforms[i].empty())
+         continue;
+
+      out->cbuffers[i].uniform_count = uniforms[i].size();
+
+      uniforms[i].push_back({ NULL });
+      out->cbuffers[i].uniforms =
+            (uniform_sem_t*)malloc(uniforms[i].size() * sizeof(*uniforms[i].data()));
+      memcpy(
+            out->cbuffers[i].uniforms, uniforms[i].data(),
+            uniforms[i].size() * sizeof(*uniforms[i].data()));
+   }
+
+   return true;
+}
+
+bool slang_process(
+      video_shader*          shader_info,
+      unsigned               pass_number,
+      enum rarch_shader_type dst_type,
+      unsigned               version,
+      const semantics_map_t* semantics_map,
+      pass_semantics_t*      out)
+{
+   Compiler*          vs_compiler = NULL;
+   Compiler*          ps_compiler = NULL;
+   video_shader_pass& pass        = shader_info->pass[pass_number];
+   glslang_output     output;
+
+   if (!glslang_compile_shader(pass.source.path, &output))
+      return false;
+
+   if (!slang_preprocess_parse_parameters(output.meta, shader_info))
+      return false;
+
+   pass.source.string.vertex   = NULL;
+   pass.source.string.fragment = NULL;
+
+   try
+   {
+      ShaderResources vs_resources;
+      ShaderResources ps_resources;
+      string          vs_code;
+      string          ps_code;
+
+      if (dst_type == RARCH_SHADER_HLSL || dst_type == RARCH_SHADER_CG)
+      {
+         vs_compiler = new CompilerHLSL(output.vertex);
+         ps_compiler = new CompilerHLSL(output.fragment);
+      }
+      else
+      {
+         vs_compiler = new CompilerGLSL(output.vertex);
+         ps_compiler = new CompilerGLSL(output.fragment);
+      }
+
+      vs_resources = vs_compiler->get_shader_resources();
+      ps_resources = ps_compiler->get_shader_resources();
+
+      if (dst_type == RARCH_SHADER_HLSL || dst_type == RARCH_SHADER_CG)
+      {
+         CompilerHLSL::Options options;
+         CompilerHLSL*         vs = (CompilerHLSL*)vs_compiler;
+         CompilerHLSL*         ps = (CompilerHLSL*)ps_compiler;
+         options.shader_model     = version;
+         vs->set_options(options);
+         ps->set_options(options);
+
+         std::vector<HLSLVertexAttributeRemap> vs_attrib_remap;
+#if 0
+         /* remaps vertex shader output too so disable for now */
```

Comment:
This should not be needed. Position and TexCoord are bound by location, so they will get TEXCOORD0, TEXCOORD1 by default. Should be possible to set that up in the input description directly.

Reply:
yeah that's what I'm doing, I thought I will just bypass the need to create a new descriptor.
I will remove that part.